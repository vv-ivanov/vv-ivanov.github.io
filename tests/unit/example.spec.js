import { shallowMount, mount, createLocalVue } from "@vue/test-utils";
import BaseFormAuth from "@/components/base/BaseFormAuth.vue";

const localVue = createLocalVue();
import "@/bootstrap.js";

describe("BaseFormAuth.vue", () => {
  it("render", () => {
    // const msg = "Вход";
    // const wrapper = shallowMount(BaseFormAuth, {
    //   propsData: { msg }
    // });
    const wrapper = mount(BaseFormAuth, {
      mocks: {
        $t: (value) => `$t('${value}')`,
      },
      localVue,
    });
    wrapper.vm.show("auth");
    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });
});
