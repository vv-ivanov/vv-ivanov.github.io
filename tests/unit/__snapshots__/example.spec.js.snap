// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseFormAuth.vue render 1`] = `
<div
  class="auth"
>
  <div
    class="auth-container"
  >
    <h5>
      $t('auth')
    </h5>
     
    <input
      class="form-control"
      placeholder="$t('valuePlaceholder')"
      type="tel"
    />
     
    <input
      autocomplete="current-password"
      class="form-control"
      placeholder="$t('passwordPlaceholder')"
      type="password"
    />
     
    <div
      class="custom-control custom-checkbox"
    >
      <input
        autocomplete="off"
        class="custom-control-input"
        type="checkbox"
        value="1"
      />
      <label
        class="custom-control-label"
      >
        
        $t('persistentAuth')
        
        <a
          href="#"
          style="float: right;"
        >
          $t('forgotPassword')
        </a>
      </label>
    </div>
     
    <button
      class="btn btn btn-primary w-100 btn-secondary disabled"
      disabled="disabled"
      type="button"
    >
      $t('buttonAuth')
    </button>
     
    <!---->
     
    <!---->
     
    <!---->
     
    <div
      class="footer"
    >
      
        $t('questionRegistration')
        
      <a
        href="#"
      >
        $t('registration')
      </a>
       
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;
